server/api/main.py

```diff
@@ -232,11 +233,11 @@ async def move_api_to_online():
         server.api.initial_data.update_default_configuration_data()
         # runs cleanup/monitoring is not needed if we're not inside kubernetes cluster
         if get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster():
-            _start_periodic_cleanup()
-            _start_periodic_runs_monitoring()
             _start_periodic_pagination_cache_monitoring()
             await _start_periodic_logs_collection()
             await _start_periodic_stop_logs()
+            _start_periodic_cleanup()
```

Comment:
why move below?

Reply:
Because we ensure run logs, we want to have a chance to collect logs for runs before they are cleaned-up

Reply:
so why runs monitoring as well? I think this is not a trivial change and we should at least discuss it. cleanup should only delete pods that were last updated over 4 hours ago so their logs should have been collected before the API starts unless something else very bad happened.

Reply:
OK, I'll leave it out of this PR, as this is not the actual improvement here.